driver-scala/src/main/scala/org/mongodb/scala/model/WindowedComputations.scala

```diff
@@ -0,0 +1,448 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mongodb.scala.model
+
+import com.mongodb.annotations.Beta
+import com.mongodb.client.model.{ MongoTimeUnit => JMongoTimeUnit, WindowedComputations => JWindowedComputations }
+
+/**
+ * Builders for [[WindowedComputation windowed computations]] used in the
+ * [[Aggregates.setWindowFields \$setWindowFields]] pipeline stage
+ * of an aggregation pipeline. Each windowed computation is a triple:
+ *  - A window function. Some functions require documents in a window to be sorted
+ *  (see `sortBy` in [[Aggregates.setWindowFields]]).
+ *  - An optional [[Window window]], a.k.a. frame.
+ *  Specifying `null` window is equivalent to specifying an unbounded window,
+ *  i.e., a window with both ends specified as [[Windows.Bound UNBOUNDED]].
+ *  Some window functions require to specify an explicit unbounded window instead of specifying `null`.
+ *  - A path to an output field to be computed by the window function over the window.
+ *
+ * A windowed computation is similar to an [[Accumulators accumulator]] but does not result in folding documents constituting
+ * the window into a single document.
+ *
+ * @see [[http://docs.mongodb.org/manual/meta/aggregation-quick-reference/#field-paths Field paths]]
+ * @since 4.3
+ * @note Requires MongoDB 5.0 or greater.
+ */
+@Beta
+object WindowedComputations {
+
+  /**
+   * Creates a windowed computation from a document field in situations when there is no builder method that better satisfies your needs.
+   * This method cannot be used to validate the document field syntax.
+   *
+   * {{{
+   *  val pastWeek: Window = Windows.timeRange(-1, MongoTimeUnit.WEEK, Windows.Bound.CURRENT)
+   *  val pastWeekExpenses1: WindowedComputation = WindowedComputations.sum("pastWeekExpenses", "$expenses", pastWeek)
+   *  val pastWeekExpenses2: WindowedComputation = WindowedComputations.of(
+   *      BsonField("pastWeekExpenses", Document("$sum" -> "$expenses",
```

Comment:
Note that the Gradle task `:driver-scala:scaladoc` produces warnings, e.g.,
```
WindowedComputations.scala:49: warning: Variable expenses undefined in comment for method of in object WindowedComputations
   *  val pastWeekExpenses1: WindowedComputation = WindowedComputations.sum("pastWeekExpenses", "$expenses", pastWeek)
```
But they do not to make sense to me, and escaping `$` as `\$` does not help as it results in `\$` being rendered in the documentation.

Reply:
Its to do with scala templating - I'll add a ticket to clean up any scaladoc warnings post 4.3

Reply:
```
   * {{{
   *  val pastWeek: Window = Windows.timeRange(-1, MongoTimeUnit.WEEK, Windows.Bound.CURRENT)
   *  val pastWeekExpenses1: WindowedComputation = WindowedComputations.sum("pastWeekExpenses", "\$expenses", pastWeek)
   *  val pastWeekExpenses2: WindowedComputation = WindowedComputations.of(
   *      BsonField("pastWeekExpenses", Document("\$sum" -> "\$expenses",
   *          "window" -> pastWeek.toBsonDocument)))
   * }}}
```

Works for me I get a code block:

```
val pastWeek: Window = Windows.timeRange(-1, MongoTimeUnit.WEEK, Windows.Bound.CURRENT)
val pastWeekExpenses1: WindowedComputation = WindowedComputations.sum("pastWeekExpenses", "$expenses", pastWeek)
val pastWeekExpenses2: WindowedComputation = WindowedComputations.of(
    BsonField("pastWeekExpenses", Document("$sum" -> "$expenses",
        "window" -> pastWeek.toBsonDocument)))
```